core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -285,6 +261,11 @@ protected String handleRequest(Channel channel, StreamInput buffer, long request
         return action;
     }
 
+    protected void validateRequest(Marker marker, StreamInput buffer, long requestId, String action) throws IOException {
+        marker.validateRequest(buffer, requestId, action);
```

Comment:
I meant take the marker logic and put it in two methods :) this way it is a pretty useless method...